src/utils/index.ts

```diff
@@ -52,15 +52,18 @@ export function each(obj: any, iterator: (value: any, key: any) => void | Breake
     }
 }
 
-export const extend = function (obj: Record<string, any>, ...args: Record<string, any>[]): Record<string, any> {
+export const extend = function <T1 extends Record<string, any>, T2 extends Record<string, any>>(
+    obj: T1,
+    ...args: T2[]
+): T1 & T2 {
```

Comment:
Minot typing nit, but we might want to revert this

This type is not generally correct, it's only true if `T1` and `T2` dont overlap. A more accurate type might be `Omit<T1, keyof T2> & T2`

The former implies that `extend({a: 5}, {a: "str"})` would return `{a: number} & {a: string}` which besides equating to `never`, it's not a reasonable type. The right return type will be just `{a: string}`

Even that is wrong because of undefined and such, but it would be slightly more correct

Second minor problem: your new type forces all `args` to have the same shape, whilst before it was much more permissible, allowing a list of objects with different types

Reply:
Haven't reviewed the "actual" code, will do so on Monday :)

Reply:
Happy to revert it, I used this for something that didn't end up making it into this PR, so it's not even useful!

Reply:
It's reverted, I made a better version of this type in a separate PR